backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/LdmlDeValidator.java

```diff
@@ -85,4 +99,81 @@ public void fatalError(SAXParseException exception) {
 
     return Norm.builder().document(result).build();
   }
+
+  /**
+   * Validate the norm against the schematron rules. Throws if a rule is not fulfilled.
+   *
+   * @param norm the norm to validate
+   * @throws IOException The schematron file couldn't be read
+   * @throws XmlProcessingException A problem occurred during the processing of the XML
+   * @throws LdmlDeSchematronException A Schematron rule is violated
+   */
+  public void validateSchematron(Norm norm) throws IOException {
+    Source xsltSource = new StreamSource(schematronXslt.getInputStream());
+    xsltSource.setSystemId(schematronXslt.getURL().toString());
+    Transformer transformer = null;
+    try {
+      transformer = new TransformerFactoryImpl().newTransformer(xsltSource);
```

Comment:
Creating a `Transformer` can be expensive. If the `schematronXslt` is a constant and doesn't change often, why not reusing a single Transformer object instead of creating one every time

You could create the object in the constructor of the `LdmlDeValidator` and ths way it will be created just once when the service is instantiated, something like this

```
    public LdmlDeValidator(@Value("classpath:schema/fixtures/legalDocML.de.xsl") Resource schematronXslt) {
        this.transformer = createTransformer(schematronXslt);
    }

    private Transformer createTransformer(Resource schematronXslt) {
        try {
            Source xsltSource = new StreamSource(schematronXslt.getInputStream());
            xsltSource.setSystemId(schematronXslt.getURL().toString());
            return new TransformerFactoryImpl().newTransformer(xsltSource);
        } catch (IOException | TransformerConfigurationException e) {
            throw new XmlProcessingException(e.getMessage(), e);
        }
    }
```